src/pydrex/vtk_helpers.py

```diff
@@ -24,34 +24,84 @@ def get_output(filename):
     return reader.GetOutput()
 
 
-def read_tuple_array(points, fieldname, dimensions):
-    """Read tuples from a vtkPointData object into a numpy array.
+def _skip_column_every(a, n):
+    """Skip a column in numpy array `a` every `n` indices.
+    
+    Examples:
 
-    Only reads the specified number of `dimensions` (columns).
-    Throws a LookupError if `fieldname` does not match any data.
+    >>> np.eye(10)
+    array([[1., 0., 0., 0., 0., 0., 0., 0., 0., 0.],
+           [0., 1., 0., 0., 0., 0., 0., 0., 0., 0.],
+           [0., 0., 1., 0., 0., 0., 0., 0., 0., 0.],
+           [0., 0., 0., 1., 0., 0., 0., 0., 0., 0.],
+           [0., 0., 0., 0., 1., 0., 0., 0., 0., 0.],
+           [0., 0., 0., 0., 0., 1., 0., 0., 0., 0.],
+           [0., 0., 0., 0., 0., 0., 1., 0., 0., 0.],
+           [0., 0., 0., 0., 0., 0., 0., 1., 0., 0.],
+           [0., 0., 0., 0., 0., 0., 0., 0., 1., 0.],
+           [0., 0., 0., 0., 0., 0., 0., 0., 0., 1.]])
+    >>> a = np.eye(10)
+    >>> _skip_column_every(a, 3)
+    array([[1., 0., 0., 0., 0., 0., 0.],
+           [0., 1., 0., 0., 0., 0., 0.],
+           [0., 0., 0., 0., 0., 0., 0.],
+           [0., 0., 1., 0., 0., 0., 0.],
+           [0., 0., 0., 1., 0., 0., 0.],
+           [0., 0., 0., 0., 0., 0., 0.],
+           [0., 0., 0., 0., 1., 0., 0.],
+           [0., 0., 0., 0., 0., 1., 0.],
+           [0., 0., 0., 0., 0., 0., 0.],
+           [0., 0., 0., 0., 0., 0., 1.]])
+    >>> _skip_column_every(a, 1)
+    array([], shape=(10, 0), dtype=float64)
+    
+    """
+    return a[:, np.mod([i for i in range(a.shape[1])], n) != n - 1]
+
+
+def read_tuple_array(points, fieldname, skip_z=False):
+    """Read tuples from VTK `points` into a numpy array.
+
+    Create a numpy array from tuples extracted from a vtkPointData object.
+
+    If `skip_z` is True, the third dimension is assumed to be full of zeros.
+    This is the standard behaviour for 2D fluidity model output.
+    In this case, indices corresponding to the third dimension are skipped.
+
+    Throws a LookupError if the `fieldname` string does not match any data.
 
     """
     data = points.GetAbstractArray(fieldname)  # Silently returns None if missing...
     if data is None:
         raise LookupError(f"unable to find data for {fieldname}")
 
-    return np.array([data.GetTuple(i) for i in range(data.GetNumberOfTuples())])[
-        :, :dimensions
-    ]
+    a = np.array([data.GetTuple(i) for i in range(data.GetNumberOfTuples())])
+    if skip_z:
+        return _skip_column_every(a, 3)
```

Comment:
Should work for tensors, from what I understand they are stored in VTK as tuples with n > dims components, so e.g. VelocityGradient is a list of tuples with 9 components each in 3D models.